### PR TITLE
XIVY-6828 Load installer data from product.json

### PR DESCRIPTION
**Pull request by alexsuter**

### CODE REVIEW

File: src/app/pages/market/ProductAction.php [hunk 1]
```diff
@@ -59,21 +63,34 @@ public function __invoke($request, $response, $args)
...
       }
+
+      $mavenArtifacts = array_filter($mavenArtifacts, fn(MavenArtifact $a) => !$a->hide());
     }
 
     $installButton = self::createInstallButton($request, $product, $version);
     
     $getInTouchLink = 'https://www.axonivy.com/marketplace/contact/?market_solutions=' . $product->getKey();
 
+    if (!empty($version)) {
```
Comment (ivy-lli) [inline src/app/pages/market/ProductAction.php:74]: why do you not check if the version is empty in the ProductMavenArtifactDownloader, like it's done in the other places (OpenAPIProvicer and ProductDescription)?

File: test/pages/market/ProductActionTest.php [hunk 2]
```diff
@@ -33,21 +33,21 @@ public function testSonatypeArtifact()
...
   public function testInstallButton_canInstallInDesignerMarket()
   {
-    AppTester::assertThatGetWithCookie('http://localhost/market/genderize', ['ivy-version' => '9.2.0'])
+    AppTester::assertThatGetWithCookie('http://localhost/market/genderize-io-connector', ['ivy-version' => '9.2.0'])
       ->ok()
-      ->bodyContains("'http://localhost/_market/genderize/_meta.json?version=");
+      ->bodyContains("'http://localhost/_market/genderize-io-connector/_product.json?version=");
   }
 
   /*public function testInstallButton_displayInDesignerMarketShowWhyNotReason()
```
Comment (ivy-lli) [inline test/pages/market/ProductActionTest.php:48]: Remove or fix test?

**Review by ivy-lli — APPROVED** [review]
Great stuff!

Some small comments below 😉

File: src/app/domain/market/Product.php [hunk 3]
```diff
@@ -272,156 +224,94 @@ public function getInstallationCount(): int
...
+    if (file_exists($productJson)) {
+      return file_get_contents($productJson);
     }
-    return "";
+    
+    return '';
   }
 
-  public function getOpenApiJson(): string
+  public function getMetaJson(): string
```
[the remark this answers is not in the document]
Comment (alexsuter) [reply src/app/domain/market/Product.php:251]: will be properley renamed later

File: src/app/RoutingRules.php [hunk 4]
```diff
@@ -74,7 +76,10 @@ public static function installRoutes($app)
     $app->get('/market', MarketAction::class);
     $app->get('/market/{key}/meta.json', MetaRedirectAction::class);
     $app->get('/market/{key}[/{version}]', ProductAction::class);
-    $app->get('/_market/{key}/_meta.json', MetaJsonAction::class);
+    $app->get('/_market/{key}/_product.json', MetaJsonAction::class);
+    $app->get('/market-cache/{key}/{version}/logo.png', MarketProductLogoRedirector::class);
```
[the remark this answers is not in the document]
Comment (alexsuter) [reply src/app/RoutingRules.php:80]: the Axon Ivy Designer is resolving the logo.png automatically relative to given product.json (ex meta.json)
for the new stuff the product.json is at the following place 
/market-cache/{key}/{version}/product.json

now the axon ivy designer automatically fetches the logo from here:
/market-cache/{key}/{version}/logo.png

to support this, I just make a redirect from
/market-cache/{key}/{version}/logo.png
to
/_market/{key}/logo.png

File: src/app/pages/market/ProductAction.php (hunk 1 above)
Comment (alexsuter) [reply src/app/pages/market/ProductAction.php:74]: will make it beautifuller later. some artifacts has no version at all. e.g. employee-on-boarding

File: test/pages/market/ProductActionTest.php (hunk 2 above)
Comment (alexsuter) [reply test/pages/market/ProductActionTest.php:48]: properly fix it later.

**Pull request merged**